src/coreclr/vm/jithelpers.cpp

```diff
@@ -2105,48 +2105,43 @@ HCIMPLEND
 //*************************************************************
 // Allocation fast path for typical objects
 //
-HCIMPL1(Object*, JIT_NewS_MP_FastPortable, CORINFO_CLASS_HANDLE typeHnd_)
+HCIMPL1_RAW(Object*, JIT_NewS_MP_FastPortable, CORINFO_CLASS_HANDLE typeHnd_)
 {
-    FCALL_CONTRACT;
-
-    do
-    {
-        _ASSERTE(GCHeapUtilities::UseThreadAllocationContexts());
+    CONTRACTL {
+        THROWS;
+        GC_NOTRIGGER;
```

Comment:
```suggestion
        DISABLED(GC_TRIGGERS);
```
GC_NOTRIGGER does not make sense - the call as a whole definitely triggers GC.